cpp/src/arrow/compute/api_vector.h

```diff
@@ -174,6 +174,23 @@ class ARROW_EXPORT SelectKOptions : public FunctionOptions {
   std::vector<SortKey> sort_keys;
 };
 
+/// \brief Fetch options
+class ARROW_EXPORT FetchOptions : public FunctionOptions {
```

Comment:
This shouldn't be a subclass of `FunctionOptions`.  `FunctionOptions` is only needed for kernel functions.  Fetch is an exec node.  Just take all these fields and put them in `FetchSinkNodeOptions` directly.